test/test_io.py

```diff
@@ -51,10 +51,10 @@ def _create_video_frames(num_frames, height, width):
 
 
 @contextlib.contextmanager
-def temp_video(num_frames, height, width, fps, lossless=False, video_codec=None, options=None):
+def temp_video(tester, num_frames, height, width, fps, lossless=False, video_codec=None, options=None):
```

Comment:
I don't think this is a good idea. This is a function that is independent on the `tester`, so I think it should stay free from it.

From my view, if you want to remove the `assert` here  (which is outside of the test btw), I'd do something like
```python
if video_codec is not None:
    raise ValueError(...)
```

Reply:
Good idea, I'll update the PR. Two quick questions:
- would you prefer a `AssertionErrorr` or `ValueError`?
- was it for error clarity that both verifications are not performed at once?